src/celengine/mapmanager.cpp

```diff
@@ -34,29 +41,28 @@ WarpMesh::~WarpMesh()
     delete data;
 }
 
-void WarpMesh::scopedDataForRendering(const function<void (float*, int)>& f) const
+std::vector<float> WarpMesh::getScopedData() const
 {
     int step = 5 * 6;
-    int size = (nx - 1) * (ny - 1) * step * sizeof(float);
```

Comment:
Appears to be a bug in the original code that allocates 4× more memory than necessary